client/js/shout.js

```diff
@@ -600,7 +613,9 @@ $(function() {
 		var button = sidebar.find(".chan[data-target=" + target + "]");
 		var isQuery = button.hasClass("query");
 		var type = msg.type;
-		var highlight = type.contains("highlight");
+		var highlight = type.contains("highlight")|| highlights.some(function(h) {
```

Comment:
Where does this `type.contains("highlight")` class or property come from?
If it's even possible, wouldn't it be better to try to apply this to the custom highlights as well, rather than patching them on the fly?